docs/HOOKS.md

```diff
@@ -1120,6 +1121,53 @@ useScrollUp(): () => void
 
 This hook will return a function that, when called, will scroll elements up the transcript. This is an important feature for AT accessibility.
 
+## `useShowModal`
+
+<!-- prettier-ignore-start -->
+```ts
+type ModalOptions = {
+  'aria-label'?: string | undefined;
+  'aria-labelledby'?: string | undefined;
+  className?: string | undefined;
+};
+
+useShowModal(): (render: () => React.Element, options?: ModalOptions | undefined) => void
```

Comment:
Since we are exposing the `useShowModal`, should we model it to receive `HTMLDialogElement.returnValue`?

Say, `useShowModal(...): Promise<string>`.

Reply:
Can we keep it void for now? This doesn't seem like an essential part of the API as React provides ways to hook into any buttons. Moreover promises are not very welcomed in callbacks in React [it seems](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/66505#discussioncomment-10411110).